lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -188,7 +188,9 @@ def parse_relationship(assoc_name, assoc_data, options)
             end
 
           polymorphic = (options[:polymorphic] || []).include?(assoc_name.to_sym)
-          hash["#{prefix_key}_type".to_sym] = assoc_data[:type] if polymorphic
+          if polymorphic
+            hash["#{prefix_key}_type".to_sym] = assoc_data.present? ? assoc_data[:type] : nil
```

Comment:
I don't understand why you're not checking polymorphic && assoc_data. Is the idea that we want to set it to nil?  so, is assoc data an empty string then? `assoc_data.empty?` would work